Flippers/src/main/java/io/doist/recyclerviewext/flippers/EmptyRecyclerFlipper.kt

```diff
@@ -0,0 +1,169 @@
+package io.doist.recyclerviewext.flippers
+
+import android.util.Log
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.IdRes
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView.AdapterDataObserver
+
+/**
+ * Handles a [RecyclerView]'s empty view, similar to the automatic control
+ * provided by [ListActivity] for [ListView].
+ */
+open class EmptyRecyclerFlipper() {
+    protected lateinit var recyclerView: RecyclerView
+    protected lateinit var emptyView: View
+    protected lateinit var flipper: Flipper
+    protected val delayedFlipper: DelayedFlipper?
+        get() = (flipper as? DelayedFlipper)
+
+    private var adapter: RecyclerView.Adapter<*>? = null
+    private val adapterObserver: AdapterDataObserver = EmptyAdapterDataObserver()
+    private var currentItemCount = 0
+
+    @Deprecated("Flipper access will be removed, customize flipper with configure method.")
```

Comment:
I've added to have some compatibility with the old class being a direct descendant of Flipper and now it does not have the Flipper methods. Do you have any better recommendation to implement this?

Reply:
I'd recommend to have `DelayedEmptyRecyclerFlipper` and keep API of `EmptyRecyclerFlipper` unchanged :)

Reply:
I'll add the Delayed versions as a separate class, but I'm maintaining the configure approach, so the flipper can be configured with a lifecycle so it can clean itself up automatically